bundles/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/handler/SamsungTvHandler.java

```diff
@@ -187,6 +206,21 @@ private synchronized void putOffline() {
         updateState(SOURCE_APP, new StringType(""));
     }
 
+    private void poll() {
+        for (SamsungTvService service : services) {
+            for (String channel : service.getSupportedChannelNames()) {
+                if (isLinked(channel)) {
```

Comment:
`isLinked` is very expensive. But since this is still an improvement over the old behaviour, I'm ok with it. The better way IMO would be to keep track of the linked channels locally.

Reply:
Why is it slow? Isn't that something that should be fixed in the core then? Why provide the method if the user shouldn't use it?

Reply:
It‘s not exactly slow but IIRC the link registry is a single map. Calling it a low frequency is not a problem. But a lot of requests with high rate could create problems, especially if a lot of bindings are doing it in larger systems.

Reply:
I just looked at the core code and it looks like it is using read locks on a single map. So it should work just fine at a high volume of reads.

Reply:
Then I'll never mention that again. :D Maybe that was refactored, I had similar code in one of my ESH bindings and was asked to remove that.